webapp/channels/src/components/move_thread_modal/move_thread_modal.tsx

```diff
@@ -0,0 +1,256 @@
+    const currentTeam = useSelector(getCurrentTeam);
+
+    const [bodyHeight, setBodyHeight] = useState<number>(0);
+    const [hasError, setHasError] = useState<boolean>(false);
+    const [postError, setPostError] = useState<React.ReactNode>(null);
+    const [selectedChannel, setSelectedChannel] = useState<ChannelOption>();
+    const [isButtonClicked, setIsButtonClicked] = useState<boolean>(false);
+
+    const bodyRef = useRef<HTMLDivElement>();
+
+    const measuredRef = useCallback((node) => {
+        if (node !== null) {
+            bodyRef.current = node;
+            setBodyHeight(node.getBoundingClientRect().height);
+        }
+    }, []);
+
+    const onHide = useCallback(() => {
+        onExited?.();
+    }, [onExited]);
+
+    const handleChannelSelect = useCallback(
+        (channel: ValueType<ChannelOption>) => {
+            if (Array.isArray(channel)) {
+                setSelectedChannel(channel[0]);
+            }
+            setSelectedChannel(channel as ChannelOption);
+        },
+        [],
+    );
+
+    // since the original post has a click handler specified we should prevent any action here
+    const preventActionOnPreview = (e: React.MouseEvent) => {
+        e.preventDefault();
+        e.stopPropagation();
+    };
+
+    const messagePreviewTitle = formatMessage({
+        id: 'move_thread_modal.preview.title',
+        defaultMessage: 'Message preview',
+    });
+
+    const previewMetaData: PostPreviewMetadata = {
+        post,
+        post_id: post.id,
+        team_name: currentTeam.name,
+        channel_display_name: originalChannel.display_name,
+        channel_type: originalChannel.type,
+        channel_id: originalChannel.id,
+    };
+
+    const notificationText = (
+        <FormattedMessage
+            id='move_thread_modal.notification.dm_or_gm'
+            defaultMessage='Moving this thread changes who has access'
+            values={{
+                strong: (x: React.ReactNode) => <strong>{x}</strong>,
+            }}
+        />
+    );
+
+    const notification = (
+        <NotificationBox
+            variant={'info'}
+            text={notificationText}
+            id={'move_thread'}
+        />
+    );
+
+    const handlePostError = (error: ClientError) => {
+        setIsButtonClicked(false);
+        setPostError(error.message);
+        setHasError(true);
+        setTimeout(() => setHasError(false), Constants.ANIMATION_TIMEOUT);
+    };
+
+    const handleSubmit = async () => {
+        setIsButtonClicked(true);
+        if (!selectedChannel) {
+            setIsButtonClicked(false);
+            return;
+        }
+
+        const channel = selectedChannel.details;
+
+        let result = await actions.moveThread(post.root_id || post.id, channel.id);
+
+        if (result.error) {
+            handlePostError(result.error);
+            return;
+        }
+
+        if (
+            channel.type === Constants.MENTION_MORE_CHANNELS &&
```

Comment:
This seems brought from `forward_post_modal.tsx`, but I am curious about what exactly this means.

Reply:
I'm not sure what MENTION_MORE_CHANNELS is but it's used throughout the forward post modal as well as the quick switcher, so I'm assuming its necessary, no noticeable regression when removed though